src/config.rs

```diff
@@ -688,6 +688,14 @@ impl AutoHeaderValue {
     }
 }
 
+impl From<Arc<String>> for AutoHeaderValue {
+    fn from(value: Arc<String>) -> Self {
+        match value.as_str() {
+            "" => Self::None,
+            _ => Self::Provided(value)
+        }
+    }
+}
```

Comment:
Would this conflict if we wanted to do a `impl<S: AsRef<str>> From<S> for AutoHeaderValue { ... }`?

Reply:
All three impls would, apparently. Would you rather the implementation be that instead of them? I figured the separate impl for `Arc<String>` for people who already have the `Arc` hanging around somewhere.

```suggestion
// impl From<Arc<String>> for AutoHeaderValue {
//     fn from(value: Arc<String>) -> Self {
//         match value.as_str() {
//             "" => Self::None,
//             _ => Self::Provided(value)
//         }
//     }
// }
// impl From<String> for AutoHeaderValue {
//     fn from(value: String) -> Self {
//         match value.as_str() {
//             "" => Self::None,
//             _ => Self::Provided(Arc::new(value))
//         }
//     }
// }
// impl From<&str> for AutoHeaderValue {
//     fn from(value: &str) -> Self {
//         match value {
//             "" => Self::None,
//             _ => Self::Provided(Arc::new(value.to_owned()))
//         }
//     }
// }
impl<S : AsRef<str>> From<S> for AutoHeaderValue {
    fn from(value: S) -> Self {
        match value.as_ref() {
            "" => Self::None,
            _ => Self::Provided(Arc::new(value.as_ref().to_owned()))
        }
    }
}
```

Reply:
Yeah. I prefer this one. If you change the PR we merge.

Reply:
Done